test/Hpack/RunSpec.hs

```diff
@@ -245,6 +259,24 @@ spec = do
           , "  default-language: Haskell2010"
           ]
 
+
+      it "requires cabal-version >= 2.0 if using generated modules" $ do
+        renderPackage_ package { packageExecutables = Map.singleton "foo" $ section $ Executable (Just "Main.hs") ["ABC"] ["ABC"]} `shouldBe` unlines [
+            "name: foo"
+          , "version: 0.0.0"
+          , "build-type: Simple"
+          , "cabal-version: >= 2.0"
+          , ""
+          , "executable foo"
+          , "  main-is: Main.hs"
+          , "  other-modules:"
+          , "      ABC"
+          , "  autogen-modules:"
+          , "      ABC"
+          , "  default-language: Haskell2010"
+          ]
+
+
```

Comment:
I would ditch this two tests.  Either we have it covered by the acceptance tests or have a function

```haskell
requiredCabalVersion :: Package -> Version
```

and unit tests for that function.